dask/dataframe/core.py

```diff
@@ -8050,3 +8109,8 @@ def _raise_if_not_series_or_dataframe(x, funcname):
             "`%s` is only supported with objects that are Dataframes or Series"
             % funcname
         )
+
+
+def _getattr_numeric_only(*args, method_name, **kwargs):
```

Comment:
This is a bit nit-picky, but hypothetically, what if `$method_name` had a keyword-arg called `method_name`? Maybe you have something reminiscent of the `median_approx` reduction, which has different methods for computing. Then you'd be hosed. I think a safer approach might be something with positional-only arguments

```suggestion
def _getattr_numeric_only(method_name, /, *args, **kwargs):
```

Reply:
Yeah, agreed. I tried something like this originally but since this method is passed to `map_paritions` / `reduction` it needs to have a signature who's first positional argument is a partition

Reply:
Ah, good point. In that case, guess I might prefer something obfuscated (but in the function signature) like `_dask_method_name`

Reply:
Is this what you're suggesting? 

```diff
diff --git a/dask/dataframe/core.py b/dask/dataframe/core.py
index 2e45c2832..98ea0d3fd 100644
--- a/dask/dataframe/core.py
+++ b/dask/dataframe/core.py
@@ -1917,7 +1917,7 @@ Dask Name: {name}, {layers}"""
                 token=token,
                 skipna=skipna,
                 axis=axis,
-                method_name=name,
+                _dask_method_name=name,
                 **numeric_only_kwargs,
             )
             return handle_out(out, result)
@@ -1929,7 +1929,7 @@ Dask Name: {name}, {layers}"""
                 skipna=skipna,
                 axis=axis,
                 split_every=split_every,
-                method_name=name,
+                _dask_method_name=name,
                 **numeric_only_kwargs,
             )
             if isinstance(self, DataFrame):
@@ -8111,6 +8111,6 @@ def _raise_if_not_series_or_dataframe(x, funcname):
         )


-def _getattr_numeric_only(*args, method_name, **kwargs):
+def _getattr_numeric_only(*args, _dask_method_name, **kwargs):
     with check_numeric_only_deprecation():
-        return getattr(M, method_name)(*args, **kwargs)
+        return getattr(M, _dask_method_name)(*args, **kwargs)
```

Reply:
Yeah, exactly, though I consider it a non-blocking suggestion